docs/cn-tree.md

```diff
@@ -0,0 +1,100 @@
+# cN Tree
+
+The cN tree is a 2-level tree comprised of a root and leaves. Leaf nodes each
+contain an exclusive range of keys and their associated values.
+
+## Properties
+
+### Tree
+
+- `fanout`: Number of leaf nodes within the tree.
+
+Fanout must be limited in order to prevent the tiny kvset problem from becoming
+unwieldy.
+
+### Node
+
+- `id`: Node ID.
+- `num_kvsets`: Number of kvsets contained with the node.
+- `kvsets`: List of kvsets.
+- `size`: Size of the node.
+
+#### Root
+
+No root specific properties.
+
+#### Leaf
+
+- `min_key`: Minimum key within the range of the node.
+- `max_key`: Maximum key within the range of the node.
+- `ref_cnt`: Reference count of the node.
+
+Together `min_key` and `max_key` are the edge keys of the node.
+
+## Edge Map
+
+The edge map is the authority on leaf nodes. It knows what nodes contain what
+ranges of keys. In order to access a leaf node, one must ask the edge map for a
+reference.
+
+## CNDB
+
+The cndb is the authority on tree shape.
+
+## Splitting Nodes
+
+When a node has grown larger than the configured node size, we will initiate a
+split. At this point we need to signify to CNDB that we are starting a leaf
+split. After taking a reference to the node, we must determine the edge keys for
+the new nodes. The left node after the split will have the `min_key` of the
+original node. The right node after the split will have the `max_key` of the
+original node. The midpoint is the middle of the original node's keyspace.
+
+At this point, we can construct the two new nodes. The two new nodes will exist
+in a void until they are ready to be committed to the edge map, but we will be
+building them with records along the way in cndb.
+
+In order to split, iterate over all kvsets. Initially, we can check to see if a
+kvset is contained entirely within either of the left node or the right node.
+If so, move it into the respective node. If not, split the kvset on the midpoint
+in in order to create 2 new kvsets in the new nodes, one for each of the nodes.
+
+## Joining Nodes
```

Comment:
To match flow of "Splitting Nodes" section, start this one with: When adjacent nodes are "too small", we will initiate a join.  ...